lib/type/boolean.ex

```diff
@@ -1,15 +1,72 @@
 defmodule Litmus.Type.Boolean do
-  @moduledoc """
-  Schema and validation for Boolean data type.
-  """
+  @moduledoc false
 
-  defstruct truthy: [true, "true"],
-            falsy: [false, "false"],
+  alias Litmus.Required
+
+  @truthy_default [true, "true"]
+  @falsy_default [false, "false"]
+
+  defstruct truthy: @truthy_default,
+            falsy: @falsy_default,
             required: false
 
   @type t :: %__MODULE__{
           truthy: [term],
           falsy: [term],
           required: boolean
         }
+
+  @spec validate_field(t, binary, map) :: {:ok, map} | {:error, binary}
+  def validate_field(type, field, data) do
+    with {:ok, data} <- Required.validate(type, field, data),
+         {:ok, data} <- truthy_falsy_validate(type, field, data) do
+      {:ok, data}
+    else
+      {:error, msg} -> {:error, msg}
+    end
+  end
+
+  @spec check_boolean_values(term, [term], [term]) :: true | nil
```

Comment:
for the return values, can they be `true` or `false`? that way you can update the signature to boolean type

Reply:
also, should this function be private? I see that the private function `truthy_falsy_validate` is using this, but I don't see other module using it